enterprise/kernel/src/main/java/org/neo4j/kernel/impl/enterprise/PropertyExistenceEnforcer.java

```diff
@@ -131,11 +131,17 @@ private void validateNode( long nodeId ) throws ConstraintValidationKernelExcept
                         }
 
                         // Check if this node has the mandatory property set
-                        if ( !propertyKeyIds.contains( constraint.propertyKey() ) )
+                        NodePropertyDescriptor descriptor = ((NodePropertyExistenceConstraint) constraint).descriptor();
+                        if ( descriptor.isComposite() )
                         {
-                            throw new NodePropertyExistenceConstraintViolationKernelException(
-                                    ((NodePropertyExistenceConstraint) constraint).label(),
-                                    constraint.propertyKey(), nodeId );
+                            for ( int propertyKey : descriptor.getPropertyKeyIds() )
```

Comment:
We intend to remove the double `getPropertyKeyId` / `getPropertyKeyIds` in the next PR.